pkg/scheduler/algorithm/priorities/interpod_affinity.go

```diff
@@ -292,14 +288,15 @@ func buildTopologyPairToScore(
 	processNode := func(i int) {
 		nodeInfo := allNodes[i]
 		if nodeInfo.Node() != nil {
+			// Unless the pod itself has affinity terms, We only need to process nodes hosting pods with affinity.
```

Comment:
Here you are filtering pods, not nodes.

`Unless the pod itself has affinity terms, we only need to process pods with affinity in the node`

Reply:
yup, fixed.